src/TheOpenNetwork/Address.h

```diff
@@ -0,0 +1,54 @@
+// Copyright © 2017-2022 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include "Data.h"
+#include "PublicKey.h"
+
+#include "Everscale/Address.h"
+
+#include <string>
+
+namespace TW::TheOpenNetwork {
+
+enum AddressTag : uint8_t {
+    BOUNCEABLE = 0x11,
+    NON_BOUNCEABLE = 0x51,
+    TEST_ONLY = 0x80,
+};
+
+class Address : public Everscale::Address {
```

Comment:
In an ideal world, we should separate Everscale and TON branches, but make a common ancestor. I didn't find any similar example in existing code, so we decided to reuse the code this way.

Reply:
> You can separate Address code logic into a set of static functions and use the standard code from both implementations; this inheritance is not justified.

The difference here is only in possible string representations, but structures of both objects are the same. TON address can be represented in "raw" and "user-friendly" forms (and more), while Everscale operate only with the raw address, but inside it's just `hash` + `workchainId`. That's why we reuse the inner structure and some address validation parts. 

Could you clarify why you think so, please? And to confirm, you mean to leave the set of functions to Everscale folder, right?

Reply:
I suppose it is better to use a separate class to use it in both everscale and ton blockchains and not set an inheritance dependency between ton and everscale